packages/devtools_app/lib/src/screens/deep_link_validation/deep_link_list_view.dart

```diff
@@ -0,0 +1,356 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+
+import 'package:devtools_app_shared/ui.dart';
+import 'package:flutter/material.dart';
+
+import '../../shared/common_widgets.dart';
+import '../../shared/primitives/utils.dart';
+import '../../shared/table/table.dart';
+import '../../shared/table/table_data.dart';
+import '../../shared/ui/colors.dart';
+import '../../shared/utils.dart';
+import 'deep_links_controller.dart';
+import 'deep_links_model.dart';
+
+enum TableViewType {
+  domainView,
+  pathView,
+  singleUrlView,
+}
+
+/// A view that display all deep links for the app.
+class DeepLinkListView extends StatefulWidget {
+  const DeepLinkListView({super.key});
+
+  @override
+  State<DeepLinkListView> createState() => _DeepLinkListViewState();
+}
+
+class _DeepLinkListViewState extends State<DeepLinkListView>
+    with ProvidedControllerMixin<DeepLinksController, DeepLinkListView> {
+  List<String> get androidVariants =>
+      controller.selectedProject.value!.androidVariants;
+
+  @override
+  void initState() {
+    super.initState();
+    callWhenControllerReady((_) {
```

Comment:
this might need to be called in `didChangeDependencies`. I'm surprised this doesn't fail at runtime because `controller` is not initialized until `initController()` is called in `didChangeDependencies`.